snuba/query/parser/expressions.py

```diff
@@ -15,28 +15,34 @@
 from snuba.query.parser.functions import parse_function_to_expr
 from snuba.query.parser.strings import parse_string_to_expr
 from snuba.util import is_function
+from snuba.query.dsl import plus, multiply
+
 
 minimal_clickhouse_grammar = Grammar(
     r"""
 # This root element is needed because of the ambiguity of the aggregation
 # function field which can mean a clickhouse function expression or the simple
 # name of a clickhouse function.
-root_element    = function_call / function_name
-expression      = function_call / simple_term
-parameters_list = parameter* (expression)
-parameter       = expression space* comma space*
-function_call   = function_name open_paren parameters_list? close_paren (open_paren parameters_list? close_paren)?
-simple_term     = quoted_literal / numeric_literal / column_name
-literal         = ~r"[a-zA-Z0-9_\.:-]+"
-quoted_literal  = "'" string_literal "'"
-string_literal  = ~r"[a-zA-Z0-9_\.:-]*"
-numeric_literal = ~r"-?[0-9]+(\.[0-9]+)?(e[\+\-][0-9]+)?"
-column_name     = ~r"[a-zA-Z_][a-zA-Z0-9_\.]*"
-function_name   = ~r"[a-zA-Z_][a-zA-Z0-9_]*"
-open_paren      = "("
-close_paren     = ")"
-space           = " "
-comma           = ","
+root_element          = function_call / function_name / top_arithm_expression
+expression            = function_call / simple_term
```

Comment:
It seems that, this way, arithmetic expression can only be directly at the root_level and they cannot be nested into functions.
This should be valid:
```
f(a * b, g(c * 3))
```
I think you should have low_pri_arithmetic also somewhere on the right side of this rule.

Reply:
Added low_pri_arithmetic to expression grammar rule.